docs/fundamentals/debugging/actors.md

```diff
@@ -121,3 +141,179 @@ If you are establishing a link definition but still see errors that indicate tha
+
+### `wash spy`
+
+As of wash 0.18, there is a new experimental debugging command called `wash spy`. This command
+allows you to see all messages sent between actors and providers, and is useful for debugging
+communication issues. To use it, you'll need to enable experimental mode for wash:
+
+```shell
+export WASH_EXPERIMENTAL=true
+```
+
+Then, you can run `wash spy` to see all messages sent between actors and providers. 
+
+```shell
+$ wash spy <actor name or ID>
+```
+
+You can pass the actual actor ID if you wish, but for ease of use, if a non-key is provided, wash
+will attempt to resolve it to an ID by checking if the `actor_name` or `call_alias`` fields from the
+actor's claims contains the given string. If more than one matches, then an error will be returned
+indicating the options to choose from. Note that this matching is case-insensitive.
+
+You can try out this command with the KV Counter example:
+
+```shell
+$ wget https://raw.githubusercontent.com/wasmCloud/examples/main/actor/kvcounter/wadm.yaml
+$ wash app put wadm.yaml
+
+Successfully put manifest
+$ wash app deploy kvcounter
+
+Deployed model
+$ wash spy kvcounter
+
+Spying on actor kvcounter
+
+[2023-06-21 11:24:32.397138 -06:00]
+From: HTTP Server              To: kvcounter                Host: NC7XZN5VJW2EOK6E565MSRY3JW2WU5ABUA6HTSPY6O7MVKGZTFCR3TDM
+
+Operation: HttpServer.HandleRequest
+Message: {
+  "method": "GET",
+  "path": "/api/counter",
+  "queryString": "",
+  "header": {
+    "host": [
+      "127.0.0.1:8081"
+    ],
+    "user-agent": [
+      "curl/7.88.1"
+    ],
+    "accept": [
+      "*/*"
+    ]
+  },
+  "body": []
+}
+
+[2023-06-21 11:24:32.407787 -06:00]
+From: kvcounter                To: Redis KeyValue Store     Host: CB6S44L4Q5IIRD4BJGQXN5TJXF4YFJXD5DOYBZOYRJP5U2GEZFKXNGD4
+
+Operation: KeyValue.Increment
+Message: {
+  "key": "counter:default",
+  "value": 1
+}
+```
+
+This output has the timestamp the invocation was observed at, the source and target of the
+invocation, and the content of the invocation itself. Wash spy will attempt to decode the invocation
+and output it to JSON, but if it is unable to, it will output the raw bytes instead.
+
+### `wash capture`
+
+As of wash 0.18, there is a new experimental debugging command called `wash capture`. This command
+enables to debug an issue after it occurs by capturing a window of all invocations in the lattice (1
+hour by default). A helpful comparison is that this is similar to many gaming tools that constantly
+capture your video while you game, and then at a press of a button you can save the last 60s for
+later usage/reply. This is useful for debugging issues that are difficult to reproduce, or for
+debugging issues that occur in production. Right now, the functionality is extremely simple, but if
+the community finds it useful, we plan on adding even more useful features to it (such as replaying
+invocation or setting up a lattice for reproducing the issue).
```

Comment:
Below we go over `wash capture replay`, so this parenthetical is confusing about whether replay support exists today or not